### ISSUE TITLE
Steam crashes at startup - Ubuntu 18.04

### ISSUE THREAD

**Issue by GirardR1006** [ValveSoftware/steam-for-linux#6366]
#### Your system information

* Steam client version (build number or date): 06/07/2019
* Distribution (e.g. Ubuntu): Ubuntu 18.04 LSTS
* Opted into Steam client beta?: No
* Have you checked for system updates?: Yes, updated 

#### Please describe your issue in as much detail as possible:
Installed Steam from the official repositories, installation goes well but then the client does not display login prompt. Here is the terminal output:

```
$ steam
Running Steam on ubuntu 18.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
[2019-07-06 20:05:59] Startup - updater built Jun 17 2019 23:31:08
Looks like steam didn't shutdown cleanly, scheduling immediate update check
[2019-07-06 20:05:59] Checking for update on startup
[2019-07-06 20:05:59] Checking for available updates...
[2019-07-06 20:05:59] Downloading manifest: client-download.steampowered.com/client/steam_client_ubuntu12
[2019-07-06 20:05:59] Download skipped: /client/steam_client_ubuntu12 version 1560817063, installed version 1560817063
[2019-07-06 20:05:59] Nothing to do
[2019-07-06 20:05:59] Verifying installation...
[2019-07-06 20:05:59] Performing checksum verification of executable files
[2019-07-06 20:06:00] Verification complete
```

I then tried to launch steam within the steam launch script ~./steam/steam.sh, output below
```
.steam/steam.sh 
Running Steam on ubuntu 18.04 64-bit
STEAM_RUNTIME is enabled automatically
Pins up-to-date!
[2019-07-06 19:40:44] Startup - updater built Jun 17 2019 23:31:08
Installing breakpad exception handler for appid(steam)/version(1560817063)
Looks like steam didn't shutdown cleanly, scheduling immediate update check
Installing breakpad exception handler for appid(steam)/version(1560817063)
[2019-07-06 19:40:44] Checking for update on startup
[2019-07-06 19:40:44] Checking for available updates...
[2019-07-06 19:40:44] Downloading manifest: client-download.steampowered.com/client/steam_client_ubuntu12
Installing breakpad exception handler for appid(steam)/version(1560817063)
[2019-07-06 19:40:44] Download skipped: /client/steam_client_ubuntu12 version 1560817063, installed version 1560817063
[2019-07-06 19:40:44] Nothing to do
[2019-07-06 19:40:44] Verifying installation...
[2019-07-06 19:40:44] Performing checksum verification of executable files
[2019-07-06 19:40:45] Verification complete
Gtk-Message: 19:40:45.109: Failed to load module "gail"
Gtk-Message: 19:40:45.109: Failed to load module "atk-bridge"
crash_20190706194044_1.dmp[3026]: Uploading dump (out-of-process)
/tmp/dumps/crash_20190706194044_1.dmp
.steam/steam.sh : ligne 712 :  3017 Erreur de segmentation  (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
```

Core dump file is here: https://framadrop.org/r/Rdl11aSrxd#H99TWTkBstpx9Bm50RNZvL4HR+YhF1Hynij8LUEd/+g=

Multiple reinstallations did not fix the issue.

I am under the Nouveau drivers, since Nvidia settings somehow break my desktop environment. CPU is an Intel Core i7 current generation, GPU is an Nvidia Quadro P1000

#### Steps for reproducing this issue:

1. Install steam from the official ubuntu repositories
2. Launch `steam` in the command line

**Comment by kisak-valve** [comment]
Hello @GirardR1006, can you check if any 32 bit OpenGL application like `glxinfo` from `mesa-utils:i386` works?

The attached minidump is a SIGSEGV in libc-2.27.so coming from i965_dri.so, when combined with your description hints that your system is an Optimus laptop.

**Comment by GirardR1006** [comment]
Hello, 

I actually reinstalled the nvidia drivers ( `nvidia-driver-430` on my distribution), which did not allowed me to log in on my session at first. I then used `prime-select intel` to see if I would at least be able to use my iGPU to run steam, and I was indeed able to login and launch steam. 

I got access to a runtime (`steam-runtime-heavy`), and I was able to log in successfully and play for a bit.
